providers/core/core.rtcdatachannel.js

```diff
@@ -124,6 +125,11 @@ RTCDataChannelAdapter.prototype.onclose = function (event) {
 RTCDataChannelAdapter.prototype.onmessage = function (event) {
   if (typeof event.data === 'string') {
     this.dispatchEvent('onmessage', {text: event.data});
+  } else if (this.channel.binaryType === 'arraybuffer' &&
+      typeof Components !== 'undefined' && !(event.data instanceof ArrayBuffer)) {
+    var view = {},
+      myData = Components.utils.cloneInto(event.data, view);
```

Comment:
I think it's worth linking to https://developer.mozilla.org/en-US/docs/Components.utils.cloneInto here.